src/quarterback/api/document.ts

```diff
@@ -99,23 +101,38 @@ export const listenStepUpdates = (
   ) => void,
   authToken: string
 ) => {
-  const listener = (event: EventSourceMessage) => {
-    if (event.data) {
-      const data = JSON.parse(event.data)
-      if (
-        typeof data.version != 'undefined' &&
-        data.steps &&
-        Array.isArray(data.steps) &&
-        data.clientIDs
-      ) {
-        dataListener(data.version, data.steps, data.clientIDs)
-      }
+  const listener = (event: MessageEvent) => {
+    const data = JSON.parse(event.data)
+
+    if (data.error && data.code) {
+      console.warn(data.error)
+      //call steps since?
+    }
+    if (
+      data['Transformer-Version'] &&
+      data['Transformer-Version'] !== getVersion()
+    ) {
+      console.warn(
+        `Warning! Manuscripts-transform (Frontend: ${getVersion()}) version is different on backend (${
+          data['Transformer-Version']
+        })})`
+      )
+    }
+    if (
+      typeof data.version != 'undefined' &&
+      data.steps &&
+      Array.isArray(data.steps) &&
+      data.clientIDs
+    ) {
+      dataListener(data.version, data.steps, data.clientIDs)
```

Comment:
Do you guys think we should move these changes to the [body-editor listener,](https://github.com/Atypon-OpenSource/manuscripts-body-editor/blob/0ed378e3b19375dae16755d89056b86cd7a6d1ae/src/useEditor.ts#L58) and handle the data properly there?

Reply:
As I see it, body editor should get data already prepared for consumption without details of how we fetched that and typecasted it. FYI I've changed the way we check for the transform version here - https://github.com/Atypon-OpenSource/manuscripts-article-editor/pull/524/files

Reply:
I'll keep this unresolved as a reminder to update the code once your PR is merged.

Reply:
Thanks